Core/Src/main.c

```diff
@@ -423,7 +423,7 @@ static void MX_SPI2_Init(void) {
     hspi2.Init.Direction = SPI_DIRECTION_2LINES;
     hspi2.Init.DataSize = SPI_DATASIZE_8BIT;
     hspi2.Init.CLKPolarity = SPI_POLARITY_LOW;
-    hspi2.Init.CLKPhase = SPI_PHASE_1EDGE;
+    hspi2.Init.CLKPhase = SPI_PHASE_2EDGE;

```

Comment:
Why are you switching the NAND flash SPI? It should be rising edge iirc

Reply:
Oops, my bad